samples/src/Microsoft.Azure.IIoT.App/src/Pages/Browser.razor

```diff
@@ -433,6 +437,10 @@
                 SamplingInterval = samplingInterval,
                 HeartbeatInterval = heartbeatInterval
             };
+            if (_publishEvent == null) {
+                _publishEvent = Task.Run(async () => await PublisherServiceEvents.NodePublishSubscribeByEndpointAsync(EndpointId,
```

Comment:
Why is Task.Run needed?

Reply:
because if you quickly select multiple node to be published in the UI  you will get multiple subscription to the same event. Because the subscription is for endpointId we need just one subscription for all the published nodes for that endpoint

Reply:
Not sure I understand why this means you call Task.Run, which just puts the registration on a task in thread pool.   What is the difference to just calling await … Subscribe() other than running outside of the rendering?   Is it blocking?

Reply:
The key is using Result (that blocks until method has completed), and I'm using Task.run to not block UI rendering